azure-pipelines.yml

```diff
@@ -45,12 +45,22 @@ steps:
     projects: '$(solutionFile)'
     arguments: '--configuration $(buildConfiguration) --no-restore -p:Version=$(assemblyVersion) -p:langversion=latest'
 
-- task: DotNetCoreCLI@2
+- script: dotnet test -c $(buildConfiguration) -v minimal --no-build --logger trx --collect:"XPlat Code Coverage" --results-directory './CodeCoverageResults'
+  displayName: 'dotnet test'
+  
+- task: PowerShell@2
+  inputs:
+    targetType: 'inline'
+    script: 'Invoke-WebRequest -Uri "https://codecov.io/bash" -OutFile codecov.sh'
+    pwsh: true
+  displayName: 'codecov: download script'
+
+- task: PowerShell@2
   inputs:
-    command: test
-    projects: '**/tests/*/*.csproj'
-    arguments: '--configuration $(buildConfiguration) -p:Version=$(assemblyVersion) --no-build'
-  displayName: 'dotnet test'  
+    targetType: 'inline'
+    script: 'bash codecov.sh -s $(Agent.BuildDirectory) -f "*coverage.cobertura.xml" -Z -t $($env:CODECOV_TOKEN)'
+    pwsh: true
+  displayName: 'codecov: upload'

```

Comment:
tests are done under RELEASE mode .. so not sure of the impact for code coverage report when the code is optimized.

Honestly (for this repo AND this is a small nuget library) I don't really care. Just making a nikpic point in case it's an issue.

Reply:
yep, i saw that we weren't building for DEBUG, so instead of adding that I'm just doing coverage for RELEASE. if we end up adding the matrix, we can modify.